builtin/providers/aws/resource_aws_s3_bucket_object.go

```diff
@@ -74,6 +75,11 @@ func resourceAwsS3BucketObject() *schema.Resource {
 				ConflictsWith: []string{"source"},
 			},
 
+			"kms_key_id": &schema.Schema{
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+
 			"etag": &schema.Schema{
 				Type: schema.TypeString,
 				// This will conflict with SSE-C and SSE-KMS encryption and multi-part upload
```

Comment:
Oooooh you're right - I had forgotten!

Okay so we have to fall back to docs or checks within Create/Update like you said.